base/abstractarraymath.jl

```diff
@@ -57,14 +57,15 @@ julia> a = reshape(Vector(1:4),(2,2,1,1))
  1  3
  2  4
 
-julia> squeeze(a,3)
+julia> squeeze(a; dims=3)
 2×2×1 Array{Int64,3}:
 [:, :, 1] =
  1  3
  2  4
 ```
 """
-function squeeze(A::AbstractArray, dims::Dims)
+squeeze(A; dims=throw(ArgumentError("the dimensions to squeeze must be specified with a `dims` keyword argument"))) = _squeeze(A, dims)
```

Comment:
You should be able to get by with a required keyword argument here:
```julia
squeeze(A; dims) = _squeeze(A, dims)
```
That will throw an `UndefKeywordError` when `dims` is not provided.

Reply:
Oh, right, I forgot we do this automatically now. Thanks!